### PR TITLE
[Review] CMake Improvements

### PR DESCRIPTION
**Pull request by aucahuasi**
- Build with g++-5.4
- Can run pytest (all tests pass)
- Update Readme.md documentation regarding the build process

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,30 +8,42 @@ The following instructions are tested on Linux and OSX systems.
...
 cd libgdf
 git submodule update --init --recursive
 ```
 
-It is recommended to setup a conda environment for the dependencies.
+Since cmake will download and build Apache Arrow you may need to install Boost C++:
+
+```bash
+$ sudo apt-get install libboost-all-dev
+```
```
Comment (scopatz) [inline README.md]: You should also be able to get the boost deps from conda-forge as well.

Comment (kkraus14) [reply README.md]: Additionally I don't think we can assume this is going to be running on a debian machine.

**Comment by aucahuasi** [comment]
@jrhemstad Actually that is a good idea, I will create a symlink to libgdf/build/src/tests/

File: README.md (hunk 1 above)
Comment (aucahuasi) [reply README.md]: Thanks for the feedback. I made a new PR https://github.com/gpuopenanalytics/libgdf/pull/64

**Comment by aucahuasi** [comment]
PD: This new PR will make cmake can generate the google tests binaries inside gtests folder (nice idea!)

https://github.com/gpuopenanalytics/libgdf/pull/64

**Pull request merged**